avro2py/codegen.py

```diff
@@ -408,6 +408,25 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
             return rewriter.visit(node)
 
 
+def resolve_schema_reference(schema: Record, schemas : List[Record]):
+    """Substitute the schema reference with the definition.""" 
+    if not schemas:
+        return
+
+    for field in schema.resolved_schema['fields']:
+        if isinstance(field['type'], str) and field['type'].find('.') >= 0:
+            ref_schema_name = field['type'].split('.')[-1]
+            is_resolved = False
+            for s in schemas:
+                if s.name == ref_schema_name:
+                    resolve_schema_reference(s, schemas)
+                    field['type'] = s.resolved_schema
+                    is_resolved = True
+                    break
+            if not is_resolved:
+                raise ValueError(f'Unable to resolve schema reference {field["type"]} in {schema.namespace}.{schema.name}')
```

Comment:
If we raise a `ValueError` here, we need to have some handling on the CLI level to let the user know what happened as opposed to the tool just failing.

Reply:
The message in the exception will do that. The `cli` catch the exception and prints the stack trace. Also other parts of the code follow the same error handling.

Are you suggesting having a custom exception raised and handle it in cli ?